crates/solver/src/driver.rs

```diff
@@ -513,18 +513,35 @@ impl Driver {
             deadline: Instant::now() + self.solver_time_limit,
             external_prices: external_prices.clone(),
         };
+
         tracing::debug!("solving auction id {}", auction.id);
         let run_solver_results = self.run_solvers(auction).await;
         for (solver, settlements) in run_solver_results {
             let name = solver.name();
 
             let mut settlements = match settlements {
                 Ok(mut settlement) => {
+                    for settlement in &settlement {
+                        tracing::debug!(
+                            %auction_id, solver_name = %name, ?settlement,
+                            "found solution",
+                        );
+                    }
+
                     // Do not continue with settlements that are empty or only liquidity orders.
+                    let settlement_count = settlement.len();
                     settlement.retain(solver_settlements::has_user_order);
+                    if settlement_count != settlement.len() {
+                        tracing::debug!(
+                            solver_name = %name,
+                            "settlement(s) filtered containing only liquidity orders",
+                        );
```

Comment:
I'm not sure it helps in this case. The issue is for our single order solvers with X settlements, if we say N got filtered out for only having liquidity orders and M got filtered out for stable price violation, then we don't know (without further analysis) why a specific settlement was filtered (and including # of settlements filtered in each step doesn't help that much IMO). That being said, if you think this number is useful for debugging, adding it is super easy! LMK.

Since external order solvers only have one settlement, this log is informative enough for them.

Reply:
> Since external order solvers only have one settlement, this log is informative enough for them.

Good point. 👍